load: existing_data at tg level with possible load overwrite

ppanero opened the pull request:
give more granularity to the existing data flags, it would support the following use cases:

- Load fully with transformation, from NON existing data
- Load fully from existing data
- Load some tg from existing data and some with trasnformation

slint commented on invenio_rdm_migrator/load/postgresql.py:
```diff
@@ -45,7 +45,13 @@ class PostgreSQLCopyLoad(Load):
+        self,
+        db_uri,
+        table_generators,
+        tmp_dir=None,
+        data_dir=None,
```
minor/comment: just to clarify:

- `tmp_dir` is used now when table generators actually populate a CSV file, and
- `data_dir` is for when we have existing data?

ppanero replied:
Yes, correct. `data_dir` is seen as permanent, a place where actual data to import is placed. `tmp_dit` is _temporary_, which you can remove after a run if you are not happy with the result for whatever reason.

slint commented on invenio_rdm_migrator/load/postgresql.py:
```diff
@@ -257,25 +281,17 @@ class ExistingDataTableGenerator(TableGenerator):
     def __init__(self, table, pks=None, post_load_hooks=None):
         """Constructor."""
-        assert not isinstance(table, list)
+        assert not isinstance(table, list) and not isinstance(table, dict)
```
```suggestion
        assert not isinstance(table, (list, dict))
```

slint reviewed (approved):
Nice! A minor question, but otherwise looks good. 

This one probably wasn't that fun to write :)

ppanero commented:
Thanks for the review! I've addressed the comment so I'm merging.

> This one probably wasn't that fun to write :)

Not really haha and I'm not 100% happy with the resulting API. It's somehow not very clear/straight forward... but #pragmatism.
